airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/destination/async/DetectStreamToFlushTest.kt

```diff
@@ -8,143 +8,129 @@ import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
 import io.airbyte.cdk.integrations.destination.async.buffers.BufferDequeue
 import io.airbyte.cdk.integrations.destination.async.function.DestinationFlushFunction
 import io.airbyte.protocol.models.v0.StreamDescriptor
+import io.mockk.every
+import io.mockk.mockk
 import java.time.Clock
 import java.time.Duration
 import java.time.Instant
 import java.util.Optional
-import java.util.concurrent.atomic.AtomicBoolean
-import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
-import org.mockito.Mockito
 
 @SuppressFBWarnings(value = ["BC_IMPOSSIBLE_CAST"])
 class DetectStreamToFlushTest {
-    val NOW: Instant = Instant.now()
-    val FIVE_MIN: Duration = Duration.ofMinutes(5)
-    private val SIZE_10MB = (10 * 1024 * 1024).toLong()
-    private val SIZE_200MB = (200 * 1024 * 1024).toLong()
-
-    private val DESC1: StreamDescriptor = StreamDescriptor().withName("test1")
+    companion object {
+        val NOW: Instant = Instant.now()
+        val FIVE_MIN: Duration = Duration.ofMinutes(5)
+        private const val SIZE_10MB = (10 * 1024 * 1024).toLong()
+        private const val SIZE_200MB = (200 * 1024 * 1024).toLong()
+        private val DESC1: StreamDescriptor = StreamDescriptor().withName("test1")
+    }
 
+    private lateinit var bufferDequeue: BufferDequeue
     private lateinit var flusher: DestinationFlushFunction
+    private lateinit var runningFlushWorkers: RunningFlushWorkers
 
     @BeforeEach
     internal fun setup() {
-        flusher = Mockito.mock(DestinationFlushFunction::class.java)
-        Mockito.`when`(flusher.optimalBatchSizeBytes).thenReturn(SIZE_200MB)
+        bufferDequeue = mockk()
+        flusher = mockk()
+        runningFlushWorkers = mockk()
```

Comment:
Did we decide on continuing with mockito in initial conversion phase or 
Is the intention to slowly start introducing mockk for DI'd classes for new path ?